xl/trax/track.py

```diff
@@ -891,11 +891,21 @@ def get_rating(self):
         :rtype: int
         """
         try:
-            rating = float(self.get_tag_raw('__rating'))
-        except (TypeError, KeyError, ValueError):
+            f = self._get_format_obj()
+            rating = None
+            if f and '__rating' in f.tag_mapping and settings.get_option('collection/write_rating_to_audio_file_metadata', False):
+                rating = self.get_tag_disk('__rating')
+            if rating:
+                rating = rating[0]
+            elif rating is None:
+                rating = self.get_tag_raw('__rating')
+            if rating is None:
+                return 0
```

Comment:
I would delete the `if rating is None` block and move `rating = float(rating)` to the end here. If `rating is None` then the result will be the same (`float(None) -> TypeError -> return 0`) and it also handles any other case where `rating` cannot be coerced into a `float`.

Reply:
i still think this is applicable. It seems like the `except TypeError` is better designed to handle this. At this point maybe just replace this with:

```py
try:
    rating = float(rating)
except TypeError:
    return 0
```

Probably don't need to catch the `TypeError`s in the above blocks either as I'm pretty sure that was only to handle the `float` conversion.

Reply:
I'm sorry if I haven't communicated this clearly but I don't think I'm getting through.

The original was:

```py
        try:
            rating = float(self.get_tag_raw('__rating'))
        except (TypeError, KeyError, ValueError):
            return 0
```

The `TypeError` was to catch the `float()` part of the statement. I think you still ought to be wrapping that type conversion in a `try` / `except TypeError` and you still ought to be wrapping the rest of the statement (`self.get_tag_raw('__rating')`) in `try` / `except (KeyError, ValueError)`.

Reply:
I think your suggestion is good.
But right know I would say, the approach is odd. Now it would read the rating from file each time the track is shown, which seems to much.
I would say, rating should only be read during scanning the library. Then this part is not necessary anymore.

Reply:
If I'm understanding you correctly, you're saying this logic should be reverted to the original and the `get_tag_disk` call should be moved to where the scan occurs and there insert the metadata into the exaile database. That makes good sense to me.